torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -823,6 +823,7 @@ void CudaCodeGen::Initialize() {
   stmt_v = stmt_v->accept_mutator(&atomic_add_fuser);
   PrioritizeLoad prioritize_load;
   stmt_v = prioritize_load.Process(stmt_v);
+  stmt_v = IRSimplifier::simplify(stmt_v);
```

Comment:
Nit: I would suggest to split an IRSimplifier fix and adding a call to it from a codegen into two separate commits (but no need to do that now).